File: fix-matchbadge.js
// fix-matchbadge.js
const fs = require('fs');
const content = fs.readFileSync('src/app/page.tsx', 'utf8');

// The match badge span spans lines 348-353
const oldBlock = `<span style={{
              fontSize: 10, fontWeight: 700, color: "#D97706",
              background: "rgba(217,119,6,0.08)", padding: "2px 7px", borderRadius: 20, flexShrink: 0,
            }}>
              {researcher.match}%
            </span>`;

const newBlock = '<MatchBadge score={researcher.match} />';

if (content.includes(oldBlock)) {
  const newContent = content.split(oldBlock).join(newBlock);
  fs.writeFileSync('src/app/page.tsx', newContent, 'utf8');
  console.log('SUCCESS: MatchBadge replaced');
} else {
  console.log('Pattern not found - checking with JSON');
  const idx = content.indexOf('fontSize: 10, fontWeight: 700, color: "#D97706"');
  if (idx >= 0) {
    console.log('Found at:', idx);
    console.log('Context:', content.substring(idx-10, idx+200));
  } else {
    console.log('Not found at all');
  }
}

File: fix_matchbadge.py
with open('src/app/page.tsx', 'r', encoding='utf-8') as f:
    content = f.read()

old = """// --- MATCH BADGE (variant-b: subtle dark - near-invisible metadata) -----
function MatchBadge({ score }: { score: number }) {
  const isHigh = score >= 90;
  const isMid  = score >= 70 && score < 90;
  const color  = isHigh ? "var(--match-high)" : isMid ? "var(--match-mid)" : "var(--match-low)";
  const bg     = isHigh ? "var(--match-high-bg)" : isMid ? "var(--match-mid-bg)" : "var(--match-low-bg)";
  return (
    <span style={{
      display: "inline-flex", alignItems: "center",
      fontSize: 10, fontWeight: 600,
      color, background: bg,
      padding: "1px 6px", borderRadius: 20,
      letterSpacing: "0.02em", flexShrink: 0,
      opacity: 0.70,
    }}>
      {score}%
    </span>
  );
}"""

new = """// --- MATCH BADGE (variant-b: subtle dark - refined metadata tier) -----
function MatchBadge({ score }: { score: number }) {
  const isHigh = score >= 90;
  const isMid  = score >= 70 && score < 90;
  const color  = isHigh ? "var(--match-high)" : isMid ? "var(--match-mid)" : "var(--match-low)";
  const bg     = isHigh ? "var(--match-high-bg)" : isMid ? "var(--match-mid-bg)" : "var(--match-low-bg)";
  const border = isHigh ? "var(--match-high-border)" : isMid ? "var(--match-mid-border)" : "var(--match-low-border)";
  return (
    <span style={{
      display: "inline-flex", alignItems: "center",
      fontSize: 10, fontWeight: 700,
      color, background: bg,
      border: `1px solid ${border}`,
      padding: "1px 6px", borderRadius: 20,
      letterSpacing: "0.02em", flexShrink: 0,
      boxShadow: isHigh ? `0 0 8px ${color}20` : "none",
    }}>
      {score}%
    </span>
  );
}"""

if old in content:
    print('FOUND')
    content = content.replace(old, new, 1)
    with open('src/app/page.tsx', 'w', encoding='utf-8') as f:
        f.write(content)
    print('REPLACED')
else:
    print('NOT FOUND')
    idx = content.find('function MatchBadge')
    if idx >= 0:
        print(repr(content[idx:idx+800]))